obspy/dyna/core.py

```diff
@@ -0,0 +1,827 @@
+    :rtype: :class:`~obspy.core.stream.Stream`
+    :return: A ObsPy Stream object.
+
+    .. rubric:: Example
+
+    >>> from obspy.core import read
+    >>> st = read("/path/to/19971014_152309ITDPC_NCR__WEX.DAT")
+    >>> st  # doctest: +ELLIPSIS
+    <obspy.core.stream.Stream object at 0x...>
+    >>> print(st)  # doctest: +ELLIPSIS
+    1 Trace(s) in Stream:
+    IT.NCR..HNE | 1970-01-01T00:00:00.000000Z - 1970-01-01T00:00:32.795000Z | 200.0 Hz, 6560 samples  # NOQA
+    """
+    headers = {}
+    data = StringIO()
+
+    # read file
+    fh = open(filename, 'rt')
+    for i in range(43):
+        key, value = fh.readline().strip().split(':')
+        headers[key.strip()] = value.strip()
+
+    # create ObsPy stream object
+    stream = Stream()
+    header = Stats()
+    header['itaca'] = {}
+
+    header['network'] = filename[-18:-16]
+    header['station'] = headers['STATION_CODE']
+    header['location'] = ''
+    if headers['COMPONENT'] == 'WE':
+        header['channel'] = 'HNE'
+    # EW should *NEVER* appear, but we handle it anyway
+    if headers['COMPONENT'] == 'EW':
+        header['channel'] = 'HNE'
+    #  -just in case ;)
+    if headers['COMPONENT'] == 'NS':
+        header['channel'] = 'HNN'
+    if headers['COMPONENT'] == 'UP':
+        header['channel'] = 'HNZ'
+    try:
+        tfs = headers['EVENT_DATE_YYYYMMDD'] + \
+            '_' + headers['TIME_FIRST_SAMPLE_S']
+        # use toUTCDateTime to convert from DYNA format
+        header['starttime'] = toUTCDateTime(tfs)
+        if re.match('^00', headers['TIME_FIRST_SAMPLE_S']) \
+               and re.match('^23', headers['EVENT_TIME_HHMMSS']):  # NOQA
+            header['starttime'] = header['starttime'] + 86400
+        if re.match('^23', headers['TIME_FIRST_SAMPLE_S']) \
+               and re.match('^00', headers['EVENT_TIME_HHMMSS']):  # NOQA
+            header['starttime'] = header['starttime'] - 86400
+    except:
+        header['starttime'] = toUTCDateTime('19700101_000000')
+    header['sampling_rate'] = 1 / float(headers['SAMPLING_INTERVAL_S'])
+    header['delta'] = float(headers['SAMPLING_INTERVAL_S'])
+    header['npts'] = int(headers['NDATA'])
+    header['calib'] = 1.0  # not in file header
+
+    # ITACA dict float data
+    header['itaca']['EVENT_LATITUDE_DEGREE'] = strtofloat(
+        headers['EVENT_LATITUDE_DEGREE'])
+    header['itaca']['EVENT_LONGITUDE_DEGREE'] = strtofloat(
+        headers['EVENT_LONGITUDE_DEGREE'])
+    header['itaca']['EVENT_DEPTH_KM'] = strtofloat(headers['EVENT_DEPTH_KM'])
+    header['itaca']['MAGNITUDE_L'] = strtofloat(headers['MAGNITUDE_L'])
+    header['itaca']['MAGNITUDE_S'] = strtofloat(headers['MAGNITUDE_S'])
+    header['itaca']['MAGNITUDE_W'] = strtofloat(headers['MAGNITUDE_W'])
+    header['itaca']['STATION_LATITUDE_DEGREE'] = strtofloat(
+        headers['STATION_LATITUDE_DEGREE'])
+    header['itaca']['STATION_LONGITUDE_DEGREE'] = strtofloat(
+        headers['STATION_LONGITUDE_DEGREE'])
```

Comment:
Not sure what you are referring to. SAC has equivalents, but Obspy `Trace` do not.
Or am I missing something?